README.md

```diff
@@ -138,6 +138,45 @@ Additional arguments can be passed to the analyzers by setting the `CLANG_TIDY_A
 
 Ccache can be enabled by configuring with `-DUSE_CCACHE=<ON | OFF>`.
 
+## Docker
+
+The project has support to Docker, that allows building and running tests inside a container.
+This container builds all targets, so you can run any of them passing a command after `sh -c`.
+
+```bash
+docker build . -t modern-cpp-starter
+
+docker run modern-cpp-starter
+```
+
+Running tests.
+
+```bash
+docker build . -t modern-cpp-starter
+
+docker run modern-cpp-starter sh -c "./build/test/GreeterTests"
+```
+
+The docker-compose can be used to manage a multi-container applications. Since this container builds all targets, you can build any specific targer passing its name after "sh -c".
+
+Running all containers.
+
+```bash
+docker-compose up --build
+```
+
+Running only your project.
+
+```bash
+docker-compose run application
+```
+
+Runing tests.
+
+```bash
+docker-compose run application sh -c "./build/test/GreeterTests"
+```
+
```

Comment:
How about bundling the commands and removing the text duplication in the docs?

``````suggestion
## Docker

The project supports Docker out-of-the-box, which allows building and running the project inside a container.
The container builds all targets, so you can run any of them passing a command after `sh -c`.

### Docker Workflow

```bash
# build
docker build . -t modern-cpp-starter
# run tests
docker run modern-cpp-starter sh -c "./build/test/GreeterTests"
# run standalone
docker run modern-cpp-starter
```

### Docker Compose Workflow

```bash
# build
docker-compose up --build
# run tests
docker-compose run application sh -c "./build/test/GreeterTests"
# run standalone
docker-compose run application
```

``````